sympy/core/function.py

```diff
@@ -1738,10 +1738,9 @@ def _eval_subs(self, old, new):
     def _eval_derivative(self, s):
         if s not in self.free_symbols:
             return S.Zero
-        return self.func(self.expr.diff(s), self.variables, self.point).doit() \
-            + Add(*[ Subs(point.diff(s) * self.expr.diff(arg),
-                    self.variables, self.point).doit() for arg,
-                    point in zip(self.variables, self.point) ])
+        return Add(*[point.diff(s)*self.func(self.expr.diff(var), self.variables, self.point).doit() + \
+                self.func(self.expr.diff(s) if s != var else S.Zero, self.variables, self.point).doit() \
```

Comment:
It looks like this term may occur several times.
```
In [42]: e = Subs(Derivative(f(x), x), (y, z), (y, z))

In [43]: print(e.diff(x))
2*Derivative(f(x), x, x)

In [45]: e = Subs(Derivative(f(x), x), (y, z, t), (y, z, t))

In [46]: print(e.diff(x))
3*Derivative(f(x), x, x)
```
The backslashes are probably not needed.